pallets/cash/src/core.rs

```diff
@@ -354,6 +416,34 @@ pub fn unapply_chain_event_internal<T: Config>(event: &ChainBlockEvent) -> Resul
                 CashPrincipalAmount(*principal),
             ),
 
+            _ => Ok(()),
+        },
+        ChainBlockEvent::Matic(_block_num, eth_event) => match eth_event {
+            ethereum_client::EthereumEvent::Lock {
+                asset,
+                sender,
+                chain,
+                recipient,
+                amount,
+            } => internal::lock::undo_lock_internal::<T>(
```

Comment:
tbh this boiler plate is a bit unnerving.  copy pasting these same calls for each chain is kind of scary